homeassistant/components/ecobee/climate.py

```diff
@@ -726,6 +739,62 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the currently active sensors on a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
+            msg = f"Invalid climate name, available options are: {', '.join(self.preset_modes)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        for sensor in sensors:
+            sensor_name = device_registry.async_get(sensor).name
+            sensor_names.append(sensor_name)
+
+        # Ensure sensors provided are available for thermostat.
+        if not set(sensor_names).issubset(set(self._sensors)):
+            msg = f"Invalid sensor for thermosat, available options are: {', '.join(self._sensors)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
+
+        # Check if sensors are currently active on thermostat.
+        current_active_sensors = self.active_sensors(climate_name)
```

Comment:
please adjust variable name to new naming ideal

Reply:
Good catch, sorry I missed that. Updated.